content/post/why-should-we-trust-ml.md

```diff
@@ -0,0 +1,74 @@
+
+As Data Science practitioners, we are creating analytical solutions to solve complex business problems. Since our output generally aids in decision making and other business decisions, it is important that these solutions are both interpretable and understandable. Since our methods are built on mathematical models, we will always benefit from increased interpretability to validate and improve our work. However, with the increase of the amount and complexity of our data has also risen the use of powerful black-box, predictive models. In this post, I will discuss techniques such as post-hoc interpretability, LIME, and SHAP, which can be employed to develop intuition behind more powerful predictive algorithms. 
+
+## Why is Interpretability Important?
+
+Consider the case of employing predictive models to use in a patient care setting. There is a lot of work going on to employ Machine Learning to predict real-time complications such as Sepsis, Renal Failure, and infections. Other uses are predicting re-admission rates, cost of the hospital stay, and general diagnosis. On average, simple, interpretable models have lower predictive models than complex, black-box models, with exceptions.
+
+![](/img/perf_vs_int.png)
+
+In this particular case, there are two sides to the same coin, of *interpretability* vs. *performance*. Advocates of performance indicate that an accurate and useful model is prferred to an understandable one, as human decision makers can be biased too. Furthermore, these models could pick up on suble variable shifts more than a human can. On the other side, advocates of interpretability indicate that if we are making clinical decisions, we must be able to explain how and why this decision was made. Patients themselves have a "right to an explanation" as to why a certain course of action was made. Furthermore, sensitive information, such as race, gender, or other variables, may be misused or incorrectly inferred by a black-box algorithm.
+
+## How can we bridge the gap?
+
+### LIME
+
+LIME approximates a black-box model with a local linear model to obtain an *individual/personalized prediction*. An explanation of an individual prediction is a local linear approximation of the models' behavior in the vicinity of that instance. 
+
+![](/img/lime.png)
+
+How does LIME work? Once your classifier is fit and you identify an instance that you want to understand, LIME samples additional data points in the neighborhood around your instance, uses the pre-fit classifier to identify which classes those samples are in, and builds a logistic regression model on all the features to develop a linear approximation to the classification around that data point. The resulting coefficients of the logistic regression add interpretation around why your data point is getting its particular prediction. LIME is completely model agnostic, and has been used with everything from image classification, text classification, and deep learning prediction.
```

Comment:
This feels really dense. I think a more helpful conversation would be an assessment of where it works well and where it doesn't work well with a link to the docs with more technical detail.